File: GreatestCommonDivisor/Java/GCD.java
class GCD {
    public static int gcd(int a, int b) {
        int temp, remainder;
        // The larger value will always be assigned to the int a.
        if (b > a) {
            temp = a;
            a = b;
            b = a;
        }
        while (b != 0) {
            remainder = a % b;
            a = b;
            b = remainder;
        }
        return a;
    }
}

File: algorithms/math/greatest-common-divisor/rust/gcd.rs
fn gcd(a: i64, b: i64) -> i64 {
    if b == 0 {
        return a;
    }
    gcd(b, a % b)
}

fn main() {
    println!("GCD of 48 and 18 is {}", gcd(48, 18));
    println!("GCD of 7 and 13 is {}", gcd(7, 13));
    println!("GCD of 0 and 5 is {}", gcd(0, 5));
}
